master/buildbot/changes/gitpoller.py

```diff
@@ -311,6 +312,16 @@ def _convert_nonzero_to_failure(res,
             "utility to handle the result of getProcessOutputAndValue"
             (stdout, stderr, code) = res
             if code != 0:
+                if code == 128:
+                    # 128 is the error code git returns when things are
+                    # wrong on its end
+                    error_message = ('RuntimeError: command %s %s in %s on '
+                                    'repourl %s failed with exit code %d: %s'
+                                    % (command, args, path, self.repourl,
+                                       code, stderr)).strip()
+                    sys.stderr.write(error_message)
+                    sys.stderr.flush()
+                    sys.exit(1)
```

Comment:
It somehow contradicts the current behaviour: if config is OK and build master started, in case of problems they just fill log file.

What about other similar components?  Two questions:
* can we detect this kind of "fatal" error?
* if yes, why do not we also stop in those cases?

Changing the behaviour for only one component feels incomplete to me.

@djmitche do we have any other considerations to take into account?